crates/bevy_ecs/src/reflect/resource.rs

```diff
@@ -176,7 +176,7 @@ impl ReflectResource {
     }
 }
 
-impl<R: Resource + FromReflect> FromType<R> for ReflectResource {
+impl<R: Resource + Reflect> FromType<R> for ReflectResource {
```

Comment:
Why did we need to change this bound?

Reply:
I'll defer to the reflection experts: I have no strong feelings here.

Reply:
As long as we rely on the fallback function this should be fine